DockerUtil.go

```diff
@@ -37,13 +33,49 @@ import (
 	"strings"
 	"syscall"
 	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ecr"
 )
 
-func StartDockerDaemon() {
-	dockerdStart := "dockerd --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &"
-	out, _ := exec.Command("/bin/sh", "-c", dockerdStart).Output()
-	log.Println(string(out))
-	waitForDockerDaemon(retryCount)
+func StartDockerDaemon(dockerConnection, dockerRegistryUrl, dockerCert string) {
+	connection := dockerConnection
+	u, err := url.Parse(dockerRegistryUrl)
+	if err != nil {
+		log.Fatal(err)
+	}
+	if connection == insecure {
+		dockerdstart := fmt.Sprintf("dockerd --insecure-registry %s --host=unix:///var/run/docker.sock --host=tcp://0.0.0.0:2375 > /usr/local/bin/nohup.out 2>&1 &", u.Host)
+		out, _ := exec.Command("/bin/sh", "-c", dockerdstart).Output()
+		logStage("Insecure Registry")
+		log.Println(string(out))
+		waitForDockerDaemon(retryCount)
```

Comment:
This logic can be extracted out in a method as same is being used in else condition